### PR TITLE
Support Django 3.1

### PR DESCRIPTION
**Pull request by adamchainz**
* Add 3.1 to the tox and travis matrices
* Run tests with deprecation and pending deprecation warnings configured to error. This catches such warnings early rather than passing them on to users.
* Fix the one Django 3.1 pending deprecation warning:

```
/.../leaflet/templatetags/leaflet_tags.py:93: RemovedInDjango40Warning: force_text() is deprecated in favor of force_str().
  layers=[(force_text(label), url, attrs) for (label, url, attrs) in instance_app_settings.get('TILES')],
```

### CODE REVIEW

File: leaflet/templatetags/leaflet_tags.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+import django
 from django import template
 from django.conf import settings
 from django.core.serializers.json import DjangoJSONEncoder
-from django.utils.encoding import force_text
 
 from leaflet import (app_settings, SPATIAL_EXTENT, SRID, PLUGINS, PLUGINS_DEFAULT,
                      PLUGIN_ALL, PLUGIN_FORMS)
 
+if django.VERSION >= (3, 0):
```
Comment (claudep) [inline leaflet/templatetags/leaflet_tags.py:11]: Isn't `force_str` supported as far as DJango 1.11?

File: tox.ini [hunk 2]
```diff
@@ -5,16 +5,18 @@ envlist =
     {py35,py36,py37}-django21,
     {py35,py36,py37}-django22,
     {py36,py37,py38}-django30,
+    {py36,py37,py38}-django31,
     {py36,py37,py38}-djangomaster
 
 [testenv]
-commands = coverage run ./quicktest.py leaflet --db={env:DATABASE:}
+commands = python -W error::DeprecationWarning -W error::PendingDeprecationWarning -m coverage run ./quicktest.py leaflet --db={env:DATABASE:}
```
Comment (claudep) [inline tox.ini:12]: I'm not necessarily against this, but I could understand that some maintainers are not super enthusiast with that choice.

File: leaflet/templatetags/leaflet_tags.py (hunk 1 above)
Comment (adamchainz) [reply leaflet/templatetags/leaflet_tags.py]: True - no need to have the version check since everything is Python 3+ here.

File: tox.ini (hunk 2 above)
Comment (Gagaro) [reply tox.ini:12]: I'm all for removing deprecated features when we can, this shouldn't be an issue.

**Comment by Gagaro** [comment]
Thanks :+1:

**Pull request merged**